File: library-domain/src/main/java/com/dodecaedro/library/domain/model/Book.java
package com.dodecaedro.library.domain.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.ZonedDateTime;
import java.util.UUID;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Book {
	@Builder.Default
	@Id
	private UUID id = UUID.randomUUID();
	private String title;
	private String isbn;
	@Column(name = "bought_date")
	private ZonedDateTime dateTimeBought;
}

File: library-domain/src/main/kotlin/com/dodecaedro/library/domain/model/Book.kt
package com.dodecaedro.library.domain.model

import java.time.ZonedDateTime
import java.util.*
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class Book(
		@Id val id: UUID? = UUID.randomUUID(),
		val title: String?,
		val isbn: String?,
		@Column(name = "bought_date") val dateTimeBought: ZonedDateTime?
)
